app/src/main/java/com/cornellappdev/android/eatery/ui/screens/EateryDetailScreen.kt

```diff
@@ -723,86 +706,34 @@ fun EateryDetailScreen(
                                                     onSelectMeal = { selectedMeal ->
                                                         eateryDetailViewModel.selectEvent(
                                                             eatery,
-                                                            weekDayIndex,
+                                                            viewState.weekdayIndex,
                                                             mealTypes[selectedMeal]
                                                         )
                                                     },
-                                                    selectedMealIndex = mealTypeIndex.value
+                                                    selectedMealIndex = viewState.mealTypeIndex
                                                 )
                                             }
                                         }
 
-                                    nextEvent.menu?.forEachIndexed { categoryIndex, category ->
-                                        val filteredItems = category.items?.filter {
-                                            it.name?.contains(filterText, true) ?: false
-                                        }
-                                        if (!filteredItems.isNullOrEmpty()) {
-                                            item {
-                                                Text(
-                                                    text = category.category ?: "Category",
-                                                    style = EateryBlueTypography.h5,
-                                                    modifier = Modifier.padding(
-                                                        horizontal = 16.dp,
-                                                        vertical = 12.dp
-                                                    )
-                                                )
+                                    menuItems(nextEvent.menu.map {
+                                        it.copy(
+                                            items = it.items.filter { menuItem ->
+                                                menuItem.item.name?.contains(filterText, true)
+                                                    ?: false
                                             }
+                                        )
+                                    }, onFavoriteClick = {
+                                        eateryDetailViewModel.toggleFavoriteMenuItem(it)
+                                    })
 
-
-                                            itemsIndexed(filteredItems) { index, menuItem ->
-                                                Row(
-                                                    verticalAlignment = Alignment.CenterVertically,
```

Comment:
No idea what you did to simplify this file, but it's a lot more concise now for sure. Good stuff!

Reply:
yeah I moved a lot of the state into the `ViewModel` and abstracted the menu to a `menuItems` extension function on `LazyListScope`. Thanks!